### ISSUE TITLE
 multi-classifications bug: Use loadModelFromHadoopFile to load the model to predict the same test set, the results are completely different

### ISSUE THREAD

**Issue by akina2016** [dmlc/xgboost#3973]
Version information 0.72, model training, and load model prediction all use xgboost4j-spark, load model using loadModelFromHadoopFile
But all predictions F1 are 0, which is completely different from the test set when training the model（don't use loadModel, use train model direct prediction F1 is 0.99）, the mode obj is multi:softprob.  How to solve it?

**Comment by akina2016** [comment]
There are bugs in 0.72 multi-classifications. All predictions of multi: soft Max are zero. When switching to multi: soft prob, they are normal and have normal predictions. But when multi: soft prob load model (xgboost4j-spark is used for training and model loading), all predictions are zero (same test set test). I don't know if it's a multi: software prob bug, or if there's a similar problem?

**Comment by akina2016** [comment]
@tqchen , Please take a look at the above questions. Thank you

**Comment by akina2016** [comment]
Looking at the following source code, we found that numclass of XGBoost Classification Model in version 0.72 is fixed value 2, i.e. two classifications. It is not read from the model. This problem has not been solved since 0.8+. This bug has been fixed and read directly from the model. That is to say, loading and forecasting of Multi-Classifications model before version 0.72 will be problematic. Would you like to pull a branch to fix the bug in version 0.72?

**Comment by hcho3** [comment]
@akina2016 In general, we do not back-port bug-fixes to older versions. Can you use 0.8+ instead?
